mmaction/datasets/pipelines/augmentations.py

```diff
@@ -42,6 +50,166 @@ def _init_lazy_if_proper(results, lazy):
         assert 'lazy' not in results, 'Use Fuse after lazy operations'
 
 
+@PIPELINES.register_module()
+class Imgaug(object):
+    """Imgaug augmentation.
+
+    Adds custom transformations from imgaug library.
+    Please, visit `https://imgaug.readthedocs.io/en/latest/index.html`
+    to get more information. An example of ``transforms`` could be found
+    in `_default_transforms`
+
+    Required keys are "imgs" and "img_shape"(if "gt_bboxes" is not None),
+    added or modified keys are "imgs", "img_shape", "gt_bboxes", "proposals".
+
+    It is worth mentioning that `Imgaug` will NOT create custom keys like
+    "interpolation", "crop_bbox", "flip_direction", etc. So when using
+    `Imgaug` along with other mmaction2 pipelines, we should pay more attetion
+    to required keys of all pipelines.
+
+    Args:
+        transforms (str or list[dict] or iaa.Augmenter): Three different ways
+            to create imgaug augmenter.
+            1) string: only support one string `default`
+            2) list[dict]: create a list of Augmenters by a list of dicts.
+                Each dict contains a key named `type`.
+            3) iaa.Augmenter: Directly use imgaug class.
+    """
+
+    def __init__(self, transforms):
+        if imgaug is None or iaa is None:
+            raise RuntimeError('imgaug is not installed')
+
+        if transforms == 'default':
+            self.transforms = self._default_transforms()
+        elif isinstance(transforms, list):
+            self.transforms = transforms
+        elif isinstance(transforms, iaa.Augmenter):
+            self.aug = self.transforms = transforms
+        else:
+            raise ValueError("transforms must be 'default' or a list of dicts")
+
+        if not isinstance(transforms, iaa.Augmenter):
+            self.aug = iaa.Sequential(
+                [self.imgaug_builder(t) for t in self.transforms])
```

Comment:
Does it mean the input transforms can also be `list[iaa.Augmenter]`?

Reply:
No, `imgaug_builder` passes a  `dict` object as input, and returns a `iaa.Augmenter` object. I'll update `imgaug_builder` docs about this.

And I don't think it's nesseary to suport `list[iaa.Augmenter]`. If we do have a `list[iaa.Augmenter]`, we can simply create a `iaa.Sequential(list_of_iaa_augmenters)` as transforms.